src/main/java/io/hops/hopsworks/expat/migrations/serving/PythonArtifactMigration.java

```diff
@@ -0,0 +1,785 @@
+      // check kubernetes is installed
+      ExpatVariables kubernetesInstalled = expatVariablesFacade.findById("kubernetes_installed");
+      if (!Boolean.parseBoolean(kubernetesInstalled.getValue())) {
+        return; // model artifacts are only created with Kubernetes installed
+      }
+    } catch (IllegalAccessException | SQLException | InstantiationException ex) {
+      String errorMsg = "Could not migrate python artifacts";
+      LOGGER.error(errorMsg);
+      throw new MigrationException(errorMsg, ex);
+    }
+  
+    PreparedStatement getProjectsStmt = null;
+    PreparedStatement getServingsStmt = null;
+    PreparedStatement updateServingWithPredStmt = null;
+    DistributedFileSystemOps dfso = null;
+    try {
+      connection.setAutoCommit(false);
+      dfso = HopsClient.getDFSO(hopsUser);
+  
+      boolean updateServings = false;
+      updateServingWithPredStmt = connection.prepareStatement(UPDATE_SERVING_WITH_PRED);
+      
+      // -- per project
+      getProjectsStmt = connection.prepareStatement(GET_PROJECTS);
+      ResultSet projectsResultSet = getProjectsStmt.executeQuery();
+      while(projectsResultSet.next()) {
+        // parse project query results
+        int projectId = projectsResultSet.getInt(1);
+        String projectName = projectsResultSet.getString(2);
+        
+        HashSet<String> keepModelArtifacts = new HashSet<>();
+        
+        // -- per serving
+        getServingsStmt = connection.prepareStatement(GET_SERVINGS);
+        getServingsStmt.setInt(1, projectId);
+        ResultSet servingsResultSet = getServingsStmt.executeQuery();
+        while(servingsResultSet.next()) {
+          // parse serving query result
+          int servingId = servingsResultSet.getInt(1);
+          String modelPath = servingsResultSet.getString(2);
+          int modelVersion = servingsResultSet.getInt(3);
+          int artifactVersion = servingsResultSet.getInt(4);
+          int modelServer = servingsResultSet.getInt(5);
+          
+          if (artifactVersion == 0) {
+            String modelName = extractModelName(projectName, modelPath);
+            if (modelServer == 0) {
+              // if tensorflow serving and artifact version 0, add this model to be ignored when deleting artifacts
+              // with version 0
+              keepModelArtifacts.add(modelName + "/" + modelVersion);
+              continue;
+            }
+            if (modelServer == 1) {
+              // if flask server
+              if (!modelPath.endsWith(".py")) {
+                // if model path does not point to a script, it's been already updated
+                keepModelArtifacts.add(modelName + "/" + modelVersion);
+                continue; // ignore serving
+              }
+              String predictor = extractPredictorFilename(modelPath);
+              // migrate artifact
+              int newArtifactVersion = migratePythonArtifact(projectName, modelName, modelVersion, predictor, dfso);
```

Comment:
Can we add a Logger.info statement here with something like "Migration artifact for projectName, modelName, modelVersion" here? In case of a failed migration it will be helpful I think. Also in any other place you think it makes sense.

Reply:
Done.